plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -15,6 +15,10 @@ IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
 	RETURN
 END
 
+DECLARE @tier AS nvarchar(128) = (SELECT TOP(1) edition from sys.database_service_objectives), 
```

Comment:
Can simplify a bit, coalesce 2 queries on sys.dm_database_replica_states into 1, also use is_local = 1 in the query to get the right replica ( will work for Azure SQL DB given singleton database). Needed in case remote replicas are ever added to this DMV. I wish at this point we could save this in a config state so don't have to run this along every single query as it adds to each query. This is something that will have to change once the new Databaseproperty lands.

	DECLARE @tier AS nvarchar(128) = (SELECT TOP(1) edition from sys.database_service_objectives)
	DECLARE @replica_id nvarchar(128), @is_primary_replica char(1)
	select @replica_id=cast(replica_id as nvarchar(128)) , 
		@is_primary_replica=cast(is_primary_replica as char(1))
		FROM sys.dm_database_replica_states
	  where is_local = 1;

Reply:
Thanks for the suggestion! Updated the query.